fdbserver/KeyValueStoreMemory.actor.cpp

```diff
@@ -307,8 +308,7 @@ class KeyValueStoreMemory final : public IKeyValueStore, NonCopyable {
 		OpSnapshotAbort, // terminate an in progress snapshot in order to start a full snapshot
 		OpCommit, // only in log, not in queue
 		OpRollback, // only in log, not in queue
-		OpSnapshotItemDelta,
-		OpEncrypted
```

Comment:
Let's keep the type but probably rename it to OpEncrypted_Deprecated, and add error log and assert in the beginning of `readOpData` if we saw it. This is to handle the case if someone enable encryption with 7.2 binary by mistake. And this is another reason why we want to keep the OpHeader format.

Reply:
Interesting point. Can they enable `encryption` with 7.2? Given the database-configuration support to control encryption is planned to be released by 7.3; I'm not sure if that could even be done. Doing by KNOBs is possible, but, we are not officially supporting encryption till `7.3`, so it is same as misconfigured cluster. 

I would vote to remove it than adding a deprecated field, unless we think the concerns raised are valid.

Reply:
A better place to assert is in `recover` when OpHeader is being read.

Reply:
I think its better to assert than to have it silently fail

Reply:
Idea is if we don't need this OP, we can remove it; am sure there would be existing code to handle `invalid` opType?

Reply:
There is but its not very clear what failed, i think this is an easier check to do that will be more clear

Reply:
@sfc-gh-ahusain we cannot enable encryption on 7.2 but the problem is that the code and the knob is there in the release. Someone could use it by mistake. We need to assert that.

Reply:
Totally agree and maybe we should have added `experimental` to the feature in the starting. 
Lets keep the OpType and maybe delete it once 7.3 is released I think?

Reply:
I think my argument is we shall never delete it. if we delete it someone else could reuse the same enum later, which may come with totally different semantic and mechanism, and so if we feed the new code with the data generate from 7.2 with OpEncrypted, the behavior is undefined, and FDB can fail in a mystery way.

We have similar issue for encrypted mutation in TLog and the encrypted redwood page, with the configurable encryption format changes. It is still undefined behavior if those code try to parse data of old format. But we are in better position there where we are reusing the same mutation type/page encoding type for the new format. When it try to parse old format it will manifest into a failure of encryption, or a segfault but a stacktrace coming from encryption, so the error is more clear.

Reply:
Makes sense. Thanks for the explanation, Yi.